src/screens/ChannelListScreenV2/IncognitoEmptyChat/index.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+import {Image, Text, View, StyleSheet, TouchableOpacity, Dimensions} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
+
+import EmptyIncognito from '../../../assets/images/empty_incognito.png';
+import IconArrowRight from '../../../assets/icons/Ic_arrow_right';
+import dimen from '../../../utils/dimen';
+import {COLORS} from '../../../utils/theme';
+import {fonts, normalizeFontSize} from '../../../utils/fonts';
+import {ANONYMOUS} from '../../../hooks/core/constant';
+import useAnonymousChannelListScreenHook from '../../../hooks/screen/useAnonymousChannelListHook';
+import {DISCOVERY_TAB_TOPICS, NavigationConstants} from '../../../utils/constants';
+
+const {width: widthScreen} = Dimensions.get('window');
+
+const MODE_FULL = 1;
+const MODE_BUTTON = 2;
+const MODE_HIDE = 3;
+
+const IncognitoEmptyChat = ({totalChannel}) => {
+  const {goToContactScreen} = useAnonymousChannelListScreenHook();
+  const navigation = useNavigation();
+
+  const getMode = () => {
+    let result = MODE_FULL;
+    if (totalChannel === 0) {
+      result = MODE_FULL;
+    }
+    if (totalChannel >= 3) {
+      result = MODE_HIDE;
+    }
+    if ((totalChannel > 0 && totalChannel < 3) || widthScreen < 380) {
+      result = MODE_BUTTON;
+    }
+
+    return result;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getMode` function is called multiple times within the component rendering. This could lead to unnecessary computations, especially if `totalChannel` or `widthScreen` values don't change often. Consider using a memoized value with `useMemo` hook for performance optimization.
```diff
-  const getMode = () => {
-    let result = MODE_FULL;
-    if (totalChannel === 0) {
-      result = MODE_FULL;
-    }
-    if (totalChannel >= 3) {
-      result = MODE_HIDE;
-    }
-    if ((totalChannel > 0 && totalChannel < 3) || widthScreen < 380) {
-      result = MODE_BUTTON;
-    }
-
-    return result;
-  };
+  const mode = React.useMemo(() => {
+    if (totalChannel === 0) {
+      return MODE_FULL;
+    }
+    if (totalChannel >= 3) {
+      return MODE_HIDE;
+    }
+    if ((totalChannel > 0 && totalChannel < 3) || widthScreen < 380) {
+      return MODE_BUTTON;
+    }
+    return MODE_FULL;
+  }, [totalChannel, widthScreen]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->